### PR TITLE
Avoid printing SETTINGS twice when formatting a query

### LINKED ISSUE #38895: Clickhouse format outputs the settings twice when format is defined
This happens when using the CLI:

```
:) select * from system.one format CSV settings max_execution_time = 0.001;

SELECT *
FROM system.one
SETTINGS max_execution_time = 0.001
FORMAT CSV
SETTINGS max_execution_time = 0.001

Query id: e9fd3179-42a3-4a0e-b205-c5444765ad26

0

1 row in set. Elapsed: 0.001 sec.
```

And also when using clickhouse format directly:

```bash
clickhouse format <<< "select * from system.one format CSV settings max_execution_time = 0.001"

SELECT *
FROM system.one
SETTINGS max_execution_time = 0.001
FORMAT CSV
SETTINGS max_execution_time = 0.001
```

As far as I can see and remember this has happened since forever (just tested 20.7 which is the oldest binary I have around and it's the same) but I couldn't find any report about it.

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Don't print `SETTINGS` more than once during query formatting if it didn't appear multiple times in the original query.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/
 
Closes https://github.com/ClickHouse/ClickHouse/issues/38895

### CODE REVIEW

**Comment by Algunenano** [comment]
@mergifyio update

**Comment by Algunenano** [comment]
@Mergifyio update

**Comment by Algunenano** [comment]
The integration tests are related to postgresql which are temporarily disabled in master (https://github.com/ClickHouse/ClickHouse/pull/39564)
The AST Fuzzer query that failed seems like a timeout, it works fine locally:
```
SELECT
    arrayResize([-2147483649, NULL], 65537),
    nullIf(number % 2, -2147483649),
    nullIf(number % 1025, 2147483646),
    100000002004087730000.,
    arrayResize(NULL, NULL)
FROM system.numbers
LIMIT 65535
FORMAT PrettyCompactNoEscapes
```

@mergifyio update

**Comment by Algunenano** [comment]
@Mergifyio update

**Comment by Algunenano** [comment]
UBSan [fuzzer failed](https://s3.amazonaws.com/clickhouse-test-reports/38900/e3ef3a2b41214093f6f2ab645abc7fe42b5fd60d/fuzzer_astfuzzerubsan//report.html) as it detected a crasher -> https://github.com/ClickHouse/ClickHouse/issues/39672
TSAN [fuzzer failed](https://s3.amazonaws.com/clickhouse-test-reports/38900/e3ef3a2b41214093f6f2ab645abc7fe42b5fd60d/fuzzer_astfuzzertsan//report.html) with `Lost connection to server. See the logs.`. I don't understand this failure; it seems like a timeout in a random query, but if that's the case then I'd expect to see way more failures as 10 seconds seems too slow of a wait for fuzzy sql. Maybe I'm not seeing something.

Not all test have finished running yet.

**Comment by Algunenano** [comment]
@Mergifyio update

File: src/Parsers/ASTSetQuery.h [hunk 1]
```diff
@@ -13,6 +13,7 @@ class ASTSetQuery : public IAST
 {
 public:
     bool is_standalone = true; /// If false, this AST is a part of another query, such as SELECT.
+    bool is_clone = false; /// If true, this AST is a clone from other part of the query and should not be printed in format()
```
Comment (rschu1ze) [inline src/Parsers/ASTSetQuery.h:16]: Minor: Maybe also mention the corresponding issue (#38895) in the comment?

Comment (rschu1ze) [inline src/Parsers/ASTSetQuery.h:16]: Minor: `printed in format()` --> `printed in FORMAT` (or did you mean `formatImpl()`?)

Comment (rschu1ze) [inline src/Parsers/ASTSetQuery.h:16]: I think the overall semantics would be more clear to the reader (e.g. in file "ASTSelectQuery.cpp") if this variable was named something like `bool print_in_format = true`.

Comment (Algunenano) [reply src/Parsers/ASTSetQuery.h:16]: Sure. It makes sense to be more explicit :+1:

Comment (Algunenano) [reply src/Parsers/ASTSetQuery.h:16]: I think it was `formatAST`. I'll improve the comment

**Review by rschu1ze — APPROVED** [review]

**Comment by rschu1ze** [comment]
Test failures are unrelated --> merging.

**Pull request merged**